Modularize provisioning hosts guide

maximiliankolb opened the pull request:
This PR splits the content of the Provisioning Hosts guide into moduluar items with one heading per file. It also touches anchors, links, and Hammer CLI parameters such as ~Your_Organization~ > `_My_Organization_`.

The built artifacts should be nearly identical.

See also #925 and #729.

maximiliankolb commented:
Ready for review 🚀

(This needs to be squashed, obviously.)

melcorr commented:
Starting here.

melcorr reviewed (changes requested):
From what I can tell, you do not need to append the "_provisioning" onto the end of links. 
The links do need a description. Only xrefs autogenerate the headline. 
I found only one weird thing. 
I think our headings are capitalised but not title case. I don't have strong feelings about blocking the PR because of this, but just an FYI. I personally like sentence case and would prefer, but I must choose my battles in life :o)

maximiliankolb commented on guides/common/assembly_provisioning-cloud-instances-openstack.adoc:
```diff
@@ -15,12 +15,13 @@ In {ProjectNameX}, you can interact with {OpenStack} REST API to create cloud in
 include::modules/snip_common-compute-resource-prereqs.adoc[]
 * A {SmartProxyServer} managing a network in your OpenStack environment.
-For more information, see {ProvisioningDocURL}Configuring_Networking[Configuring Networking].
+For more information, see {ProvisioningDocURL}Configuring_Networking_provisioning[] in the _Provisioning_ guide.
```
I'm adding the "resolved" context because the anchor is defined as `guides/common/modules/con_configuring-networking.adoc:1:[id="Configuring_Networking_{context}"]`. As we link to an anchor in a different guide with a different context, we need to add `_provisioning`.

melcorr commented on guides/common/modules/proc_using-vmware-cloud-init-and-userdata-templates-for-provisioning.adoc:
```diff
@@ -0,0 +1,198 @@
+[options="nowrap" subs="+quotes"]
+----
+# logrotate -f /etc/logrotate.conf
+# rm -f /var/log/*-???????? /var/log/*.gz
```
I had a comment here. I don't know where it went. 
Can you test locally? When I build, the asterisks here disappears ` /var/log/*.gz` becomes ` /var/log/.gz`

melcorr replied:
@maximiliankolb this was the weird thing I mentioned in my overall feedback

maximiliankolb replied:
Fixed it.

maximiliankolb commented on guides/common/assembly_provisioning-virtual-machines-kvm.adoc:
```diff
@@ -20,7 +20,7 @@ You can use KVM provisioning to create hosts over a network connection or from a
 * A {SmartProxyServer} managing a network on the KVM server.
 Ensure no other DHCP services run on this network to avoid conflicts with {SmartProxyServer}.
-For more information about network service configuration for {SmartProxyServer}s, see {ProvisioningDocURL}Configuring_Networking[Configuring Networking].
+For more information about network service configuration for {SmartProxyServer}s, see {ProvisioningDocURL}Configuring_Networking_provisioning[] in the _Provisioning_ guide.
```
My bad. I've fixed this for all instances: `git grep -i -n "DocURL}" | grep "\[\]"`

maximiliankolb commented:
Rebased and squashed.

I've added descriptions to add links with `...DocURL}[]` and fixed the code formatting issue with `*` in a code block.

Regarding title case: I use https://titlecaseconverter.com/, so my changes were not by accident. I can revert if there's a convention in IBM Style guide.

melcorr commented:
> > Regarding title case: I use https://titlecaseconverter.com/, so my changes were not by accident. I can revert if there's a convention in IBM Style guide.
> 
> I looked it up: IBM Style Guide, chapter 1 "Capitalization in headings and titles":
> 
> > In general, use sentence-style capitalization for headings.
> 
> cry

This is not followed in Satellite legacy docs, so i do not expect you to fix this. 
I'm fine with the titles. I can give the standardization of titles as a new hire exercise to someone.

maximiliankolb commented:
Awesome.

melcorr commented:
I pulled the latest changes and cannot build the guide locally.
Can you test this for me on your end @maximiliankolb ?

maximiliankolb commented:
Rebased to master. I can build locally just fine using the container image:

    rm -rf guides/build && docker run --rm -v $(pwd):/foreman-documentation foreman_documentation make html

melcorr reviewed (approved):
Issue was on my side. 
I did a clean and managed to fix. Verified with lzap !
All looks great.

melcorr commented:
I am cherry-picking these modularization PRs to 3.1 because this will be a disaster otherwise when docs team add 3.1 changes later:

   42b858c5..70b71973  3.1 -> 3.1

maximiliankolb commented:
Fantastic! I am now rebasing "managing hosts" onto master.
